From 98b80ebae3fb1c105ecf378c73b9f9a58f97fccd Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Tue, 7 May 2019 20:36:21 -0600
Subject: [PATCH 10/10] [SQUASH][DNM] Merge tag 'android-9.0.0_r37' into
 staging/lineage-16.0_merge-android-9.0.0_r37

Android 9.0.0 Release 37 (PQ3A.190505.002)

* tag 'android-9.0.0_r37':
  Preventing padding from getting reset when the widget updates
  Revert "Use clear-all recent tasks call"
  Kill launcher when display size changes
  Use clear-all recent tasks call
  Fix missing callback in fallback activity to re-enable high res loader.
  Update launcher grid for tablet devices.
  Fix gap between notification dots and their shadows
  Add null check before playing recents scale down anim
  Defer starting the high-res thumbnail loader until quick step/scrub
  Increase max distance for folder creation for tablets.
  App menu icon launching app info hides taskview menu

https://github.com/haggertk/android_packages_apps_Trebuchet/tree/lineage-16.0_merge-android-9.0.0_r37

Change-Id: I73e57c71349e3fcb7c56b89b74fc2ffdbcb45393
---
 src/com/android/launcher3/qsb/QsbContainerView.java  | 1 -
 src/com/android/launcher3/qsb/QsbWidgetHostView.java | 5 +++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/qsb/QsbContainerView.java b/src/com/android/launcher3/qsb/QsbContainerView.java
index c9f6721ef..f103f804a 100644
--- a/src/com/android/launcher3/qsb/QsbContainerView.java
+++ b/src/com/android/launcher3/qsb/QsbContainerView.java
@@ -191,7 +191,6 @@ public class QsbContainerView extends FrameLayout {
                         .getAppWidgetOptions(widgetId), opts)) {
                     mQsb.updateAppWidgetOptions(opts);
                 }
-                mQsb.setPadding(0, 0, 0, 0);
                 mQsbWidgetHost.startListening();
                 return mQsb;
             }
diff --git a/src/com/android/launcher3/qsb/QsbWidgetHostView.java b/src/com/android/launcher3/qsb/QsbWidgetHostView.java
index 906edf92a..de9caf4f5 100644
--- a/src/com/android/launcher3/qsb/QsbWidgetHostView.java
+++ b/src/com/android/launcher3/qsb/QsbWidgetHostView.java
@@ -58,6 +58,11 @@ public class QsbWidgetHostView extends AppWidgetHostView {
         return mPreviousOrientation != orientation;
     }
 
+    @Override
+    public void setPadding(int left, int top, int right, int bottom) {
+        // Prevent the base class from applying the default widget padding.
+        super.setPadding(0, 0, 0, 0);
+    }
 
     @Override
     protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
-- 
2.17.1

